app/views/granules/_generic_download.sh.erb

```diff
@@ -11,7 +11,7 @@ setup_auth_curl() {
     status=$(curl -s -z "$(date)" -w %{http_code} <%= raw @first_link %> | tail -1)
     if [[ "$status" -ne "200" && "$status" -ne "304" ]]; then
         <% # Check if this is an FTP retrieval %>
-        <% if @first_link.start_with? 'ftp' %>
+        <% if (@first_link) && (@first_link.start_with? 'ftp') %>
```

Comment:
```if @first_link && @first_link.start_with?('ftp')```
 - https://github.com/bbatsov/ruby-style-guide

Reply:
Ahhh - tried that before, but I didn't have parenthesis around the 'ftp' and it threw an exception - the way you did it there works well.  Updated ;)